diff --git a/build.info b/build.info
index 3dda4e8..ff55680 100644
--- a/build.info
+++ b/build.info
@@ -33,7 +33,7 @@ IF[{- defined $target{shared_defflag} -}]
 
     SHARED_SOURCE[libcrypto]=libcrypto.def
     SHARED_SOURCE[libssl]=libssl.def
-  ELSIF[{- $config{target} =~ /^aix/ -}]
+  ELSIF[{- $config{target} =~ /^(aix|pase)/ -}]
     GENERATE[libcrypto.map]=util/mkdef.pl crypto aix
     DEPEND[libcrypto.map]=util/libcrypto.num
     GENERATE[libssl.map]=util/mkdef.pl ssl aix
